app/scripts/controllers/newOverview.js

```diff
@@ -0,0 +1,1200 @@
+  // alternate backend).
+  //
+  // Updates `state.routesByService`
+  //   key: service name
+  //   value: array of routes, sorted by RoutesService.sortRoutesByScore
+  var groupRoutes = function() {
+    var routesByService = RoutesService.groupByService(overview.routes, true);
+    state.routesByService = _.mapValues(routesByService, RoutesService.sortRoutesByScore);
+  };
+
+  // Group HPAs by the object they scale.
+  //
+  // Updates `state.hpaByResource`
+  //   key: hpaByResource[kind][name]
+  //   value: array of HPA objects
+  var groupHPAs = function() {
+    state.hpaByResource = HPAService.groupHPAs(overview.horizontalPodAutoscalers);
+  };
+
+  // Adds a recent pipeline build to the following maps:
+  //
+  // `state.recentPipelinesByBuildConfig``
+  //   key: build config name
+  //   value: array of pipeline builds
+  //
+  // `overview.recentPipelinesByDeploymentConfig`
+  //   key: deployment config name
+  //   value: array of pipeline builds
+  var groupPipelineByDC = function(build) {
+    var bcName = getBuildConfigName(build);
+    var buildConfig = overview.buildConfigs[bcName];
+    if (!buildConfig) {
+      return;
+    }
+
+    overview.recentPipelinesByBuildConfig[bcName] = overview.recentPipelinesByBuildConfig[bcName] || [];
+    overview.recentPipelinesByBuildConfig[bcName].push(build);
+
+    // Index running pipelines by DC name.
+    var dcNames = BuildsService.usesDeploymentConfigs(buildConfig);
+    _.each(dcNames, function(dcName) {
+      state.recentPipelinesByDeploymentConfig[dcName] = state.recentPipelinesByDeploymentConfig[dcName] || [];
+      state.recentPipelinesByDeploymentConfig[dcName].push(build);
+    });
+    updatePipelineOtherResources();
+  };
+
+  // Group build configs by their output image. This lets us match them to
+  // deployment config image change triggers.
+  var buildConfigsByOutputImage = {};
+  var groupBuildConfigsByOutputImage = function() {
+    buildConfigsByOutputImage = BuildsService.groupBuildConfigsByOutputImage(overview.buildConfigs);
+  };
+
+  var getBuildConfigsForDeploymentConfig = function(deploymentConfig) {
+    var uid = getUID(deploymentConfig);
+    if (!uid) {
+      return;
+    }
+
+    return _.get(state, ['buildConfigsByObjectUID', uid], []);
+  };
+
+  // Find all recent builds for `deploymentConfig` from each of `buildConfigs`.
+  //
+  // Updates `state.recentBuildsByDeploymentConfig`
+  //   key: deployment config name
+  //   value: array of builds, sorted in descending order by creation date
+  var updateRecentBuildsForDeploymentConfig = function(deploymentConfig) {
+    var builds = [];
+    var buildConfigs = getBuildConfigsForDeploymentConfig(deploymentConfig);
+    _.each(buildConfigs, function(buildConfig) {
+      var recentForConfig = _.get(state, ['recentBuildsByBuildConfig', buildConfig.metadata.name], []);
+      builds = builds.concat(recentForConfig);
+    });
+    builds = BuildsService.sortByBuildNumber(builds, true);
```

Comment:
You're right. I was looking at a different part of the code because my numbers locally no longer line up.

But this is only used for build counts, so shouldn't need to be sorted at all.